yarn-project/circuits.js/src/keys/derivation.ts

```diff
@@ -41,9 +41,45 @@ export function deriveSigningKey(secretKey: Fr): GrumpkinScalar {
   return sha512ToGrumpkinScalar([secretKey, GeneratorIndex.IVSK_M]);
 }
 
-export function computeAddress(publicKeysHash: Fr, partialAddress: Fr) {
-  const addressFr = poseidon2HashWithSeparator([publicKeysHash, partialAddress], GeneratorIndex.CONTRACT_ADDRESS_V1);
-  return AztecAddress.fromField(addressFr);
+export function computePreaddress(publicKeysHash: Fr, partialAddress: Fr) {
+  return poseidon2HashWithSeparator([publicKeysHash, partialAddress], GeneratorIndex.CONTRACT_ADDRESS_V1);
+}
+
+export function computeAddress(publicKeys: PublicKeys, partialAddress: Fr) {
+  const preaddress = computePreaddress(publicKeys.hash(), partialAddress);
+  const address = computeAddressFromPreaddressAndIvpkM(preaddress, publicKeys.masterIncomingViewingPublicKey);
+
+  return address;
+}
+
+export function computeAddressFromPreaddressAndIvpkM(preaddress: Fr, ivpkM: Point) {
+  const addressPoint = computeAddressPointFromPreaddressAndIvpkM(preaddress, ivpkM);
+
+  return AztecAddress.fromField(addressPoint.x);
+}
+
+export function computeAddressPointFromPreaddressAndIvpkM(preaddress: Fr, ivpkM: Point) {
+  const preaddressPoint = derivePublicKeyFromSecretKey(new Fq(preaddress.toBigInt()));
```

Comment:
Using `derivePublicKeyFromSecretKey` here where in reality we just want to mul by the generator broke my brain a bit. I know it's the same operation, but seems hard to read...maybe a comment showing the computation steps?